src/v/cluster/partition_balancer_backend.cc

```diff
@@ -177,7 +177,11 @@ void partition_balancer_backend::on_members_update(
         }
     }
 
-    if (state == model::membership_state::draining) {
+    if (
+      state == model::membership_state::draining
+      // Only schedule tick on node addition if health report is already
+      // available
+      || (state == model::membership_state::active || _health_monitor.contains_node_health_report(id))) {
```

Comment:
Should this be a &&?

Also, nit, but IMO this deserves a separate if statement :)

Reply:
definitely the second one should be a `&&`